cupsfilters/ppdgenerator.c

```diff
@@ -1978,47 +1978,93 @@ ppdCreateFromIPP2(char         *buffer,          /* I - Filename buffer */
     is_pdf = 1;
   }
 #ifdef CUPS_RASTER_HAVE_APPLERASTER
-  if (cupsArrayFind(pdl_list, "image/urf")) {
+  else if (cupsArrayFind(pdl_list, "image/urf")) {
+    int resStore = 0; /* Variable for storing the no. of resolutions in the resolution array */
+    int resArray[__INT16_MAX__]; /* Creating a resolution array supporting a maximum of 32767 resolutions */
+    int lowdpi = 0, middpi = 0, hidpi = 0; /* Lower , middle and higher resolution */
     if ((attr = ippFindAttribute(response, "urf-supported",
 				 IPP_TAG_KEYWORD)) != NULL) {
-      int lowdpi = 0, hidpi = 0; /* Lower and higher resolution */
       for (i = 0, count = ippGetCount(attr); i < count; i ++) {
-	const char *rs = ippGetString(attr, i, NULL); /* RS value */
-	if (_cups_strncasecmp(rs, "RS", 2))
-	  continue;
-	lowdpi = atoi(rs + 2);
-	if ((rs = strrchr(rs, '-')) != NULL)
-	  hidpi = atoi(rs + 1);
-	else
-	  hidpi = lowdpi;
-	break;
+        const char *rs = ippGetString(attr, i, NULL); /* RS value */
+        const char *rsCopy = ippGetString(attr, i, NULL); /* RS values(copy) */
```

Comment:
It seems odd that ippGetString() is called twice when pointer assignment of  `rsCopy = rs` would have worked. I think `rsCopy` is also redundant, more about that below.